fast_ms/src/diff/compress_ms.cpp

```diff
@@ -0,0 +1,913 @@
+    histo_t counter;
+    size_type n_runs = fill_encoder<enc_type>(permutedMS, encoder, counter);
+    vec_type c_ms(encoder, permutedMS.size());	
+	char buffer[100]; 
+	sprintf(buffer,"%s/compressedMS-diff-%d-%d-%d",outputDirectory.c_str(),threshold,allowNegativeMS,greedyStrategy);
+    std::ofstream out{  buffer   /*ms_path + ms_compression::to_str(flags.compression)*/, std::ios::binary};
+    c_ms.writeTo(out);
+
+    (cerr << n_runs << " runs over "
+          << c_ms.getSize() << " elements ("
+          << c_ms.getSize() / static_cast<float>(n_runs) << " elements / run)"
+          << endl);
+    //for(auto item : counter)
+    //    cout << item.first << "," << item.second << endl;
+    return diff_from(from);
+}
+
+
+int main(int argc, char **argv){
+    OptParser input(argc, argv);
+    string ms_path;
+
+    string help__ms_path{"\t-ms_path <path to a binary>: the ms bit vector\n"};
+    if(argc == 1){
+        (cerr << "Compress a ms vector. Creates files <ms_path>.xxx\n"
+              << "Args:\n"
+              << help__ms_path
+              << help__compression
+              << endl);
+        exit(0);
+    }
+    InputFlags flags;
+    try{
+        flags = InputFlags(input);
+    }
+    catch (string s) {
+        cerr << s << endl;
+        return 1;
+    }
+    ms_path = input.getCmdOption("-ms_path");
+    cerr << "check: " << diff_from(abs_point()) << endl;//size_type mem_mark = abs_point();
+    switch(flags.compression)
+    {
+        case Compression::hybrid:
+            cout << comp<sdsl::hyb_vector<>>(ms_path, flags) << endl;
+            break;
+        case Compression::rrr:
+            cout << comp<sdsl::rrr_vector<>>(ms_path, flags) << endl;
+            break;
+        case Compression::rle:
+			cout << comp1<CSA::RLEVector, CSA::RLEEncoder> ( ms_path,flags, 
+															 input.getCmdOption("-outputDir"),
+															 input.getCmdOption("-tablesDir"),
+			                                                 (uint32_t)std::stoi(input.getCmdOption("-threshold")),
+															 (size_type)std::stoi(input.getCmdOption("-nZeros")),
+															 (size_type)std::stoi(input.getCmdOption("-nOnes")),
+														     (uint8_t)std::stoi(input.getCmdOption("-negative")),
+															 (uint8_t)std::stoi(input.getCmdOption("-greedy")),
+														     (uint8_t)std::stoi(input.getCmdOption("-verbose"))
+														   ) << endl;
+			break;
+        case Compression::delta:
+            //cout << comp1<CSA::DeltaVector, CSA::DeltaEncoder>(ms_path,flags,0,0,0,0) << endl;
+            break;
+        case Compression::succint:
+            //cout << comp1<CSA::SuccinctVector, CSA::SuccinctEncoder>(ms_path,flags,0,0,0,0) << endl;
+            break;
+        case Compression::nibble:
+            //cout << comp1<CSA::NibbleVector, CSA::NibbleEncoder>(ms_path,flags,0,0,0,0) << endl;
+            break;
+        case Compression::corr:
+            //cout << comp2(ms_path, flags) << endl;
```

Comment:
so `comp2(...)` is also not needed

Reply:
i agree. see above. i just kept all of those because they belonged to an example program you sent me a long time ago, which i expanded with my functions.